src/main/java/io/debezium/connector/yugabytedb/YugabyteDBOffsetContext.java

```diff
@@ -146,34 +145,20 @@ public static YugabyteDBOffsetContext initialContext(YugabyteDBConnectorConfig c
     @Override
     public Map<String, ?> getOffset() {
         Map<String, Object> result = new HashMap<>();
-        if (sourceInfo.timestamp() != null) {
-            result.put(SourceInfo.TIMESTAMP_USEC_KEY, Conversions
-                    .toEpochMicros(sourceInfo.timestamp()));
-        }
-        if (sourceInfo.txId() != null) {
-            result.put(SourceInfo.TXID_KEY, sourceInfo.txId());
-        }
-        if (sourceInfo.lsn() != null) {
-            result.put(SourceInfo.LSN_KEY, sourceInfo.lsn().toSerString());
-        }
-        if (sourceInfo.xmin() != null) {
-            result.put(SourceInfo.XMIN_KEY, sourceInfo.xmin());
-        }
-        if (sourceInfo.isSnapshot()) {
-            result.put(SourceInfo.SNAPSHOT_KEY, true);
-            result.put(SourceInfo.LAST_SNAPSHOT_RECORD_KEY, lastSnapshotRecord);
-        }
-        if (lastCompletelyProcessedLsn != null) {
-            result.put(LAST_COMPLETELY_PROCESSED_LSN_KEY, lastCompletelyProcessedLsn.toSerString());
-        }
-        if (lastCommitLsn != null) {
-            result.put(LAST_COMMIT_LSN_KEY, lastCommitLsn.toSerString());
+
+        for (Map.Entry<String, SourceInfo> entry : this.tabletSourceInfo.entrySet()) {
```

Comment:
Is this storing lsn for every tablet instead of fields in SourceInfo?

Reply:
Yes.